engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -58,6 +60,9 @@ object JesBackend {
 
   def authGcsCredentialsPath(gcsPath: String): JesInput = JesLiteralInput(ExtraConfigParamName, gcsPath)
 
+  // Strictly speaking this could cause OOM issues but should be fine. Something to monitor
+  val GcsUploadThreadPool = ExecutionContext.fromExecutor(Executors.newCachedThreadPool())
```

Comment:
Honestly I think it's belt & suspenders as @mcovarr would say. I initially had the separate pool prior to using the actor based retry and debated removing it. 

I left it in since it's the actual upload futures in that thread pool (at least that was my intention) and the operations in the futures do block.